src/Screens/Bookmark/BookmarkIcon.js

```diff
@@ -0,0 +1,74 @@
+import React, { useState, useEffect } from "react";
+import { View } from "react-native";
+import { useDispatch, useSelector } from "react-redux";
+import { addBookmark, removeBookmark } from "../../Redux/bookmarkSlice";
+import Icon from "react-native-vector-icons/FontAwesome";
+import styles, { Color } from "../../Shared/styles";
+import { RoutingItems } from "../../Data/RoutingLookups";
+
+export const BookmarkIcon = ({
+  title,
+  sectionNum,
+  partLabel,
+  sectionHeading,
+}) => {
+  const [marked, setMarked] = useState(false); //to change marked state of bookmark
+  const bookmarkStateId = useSelector(state => state.bookmarks.bookmarkArray); //get list of bookmarks
+  const dispatch = useDispatch();
+
+  //switch the mark state to opposite what it was.
+  function switchMarks() {
+    setMarked(!marked);
+    return marked;
+  }
+
+  //if the bookmarks exists in redux, then switch the mark upon any change of all of the elements below.
+  useEffect(() => {
+    // compares state array to see if section exists in bookmarks, if it does turn on bookmark icon
+    bookmarkStateId.some(e => e.sectionNum == sectionNum && e.title == title)
```

Comment:
I'm not super familiar with .some(), but it sounds like it's counter productive to your && operator; should it be || instead?

Also, these should probably be strict equality.

Reply:
I've changed it to strict equality and to a .find but it seems to work the same way.  Both elements need to be true for the icon to logically change.

Reply:
Okay, then .some() was definitely not the correct operator to use if you need it to match all. .some() sounds like it would accept either option (I know, I know, that's why you used the && operator...)

Reply:
It was a copy/paste from Project Nemo.  It works; who am I to argue.. hahaha